Plugin/Tax/Config/ShippingTaxPlugin.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FireGento\MageSetup\Plugin\Tax\Config;
+
+use FireGento\MageSetup\Model\System\Config\Source\Tax\Dynamic as FireGentoSource;
+use Magento\Customer\Model\ResourceModel\GroupRepository;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\Quote\Item;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\Store;
+use Magento\Tax\Api\Data\TaxClassKeyInterface;
+use Magento\Tax\Model\Calculation\Proxy;
+
+/**
+ * Class ShippingTaxPlugin
+ *
+ * FireGento\MageSetup\Plugin\Tax\Config
+ */
+class ShippingTaxPlugin
+{
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * @var GroupRepository
+     */
+    private $groupRepository;
+
+    /**
+     * @var Proxy
+     */
+    private $taxCalculation;
+
+    /**
+     * @var \Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory
+     */
+    private $taxClassKeyDataObjectFactory;
+
+    /**
+     * Constructor class
+     *
+     * @param ScopeConfigInterface                              $scopeConfig
+     * @param Session                                           $customerSession
+     * @param GroupRepository                                   $groupRepository
+     * @param \Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory $taxClassKeyDataObjectFactory
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        Session $customerSession,
+        GroupRepository $groupRepository,
+        \Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory $taxClassKeyDataObjectFactory
+    ) {
+        $this->scopeConfig                  = $scopeConfig;
+        $this->customerSession              = $customerSession;
+        $this->groupRepository              = $groupRepository;
+        $this->taxClassKeyDataObjectFactory = $taxClassKeyDataObjectFactory;
+    }
+
+    /**
+     * After plugin for \Magento\Tax\Model\Sales\Total\Quote\CommonTaxCollector::getShippingDataObject
+     *
+     * @param                                          $subject
+     * @param                                          $result
+     * @param ShippingAssignmentInterface              $shippingAssignment
+     * @param \Magento\Quote\Model\Quote\Address\Total $total
+     * @param                                          $useBaseCurrency
+     *
+     * @return null
+     *
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function afterGetShippingDataObject(
+        $subject,
+        $result,
+        ShippingAssignmentInterface $shippingAssignment,
+        \Magento\Quote\Model\Quote\Address\Total $total,
```

Comment:
Please import this class.

Reply:
class was refactored so this line is gone, but I think I imported all dependencies